### PR TITLE
Add multi-file support to `dask_cudf.read_json`

### PR DESCRIPTION
**Pull request by rjzamora**
## Description
Dask cuDF often benefits from a larger partition sizes than pandas-backed Dask DataFrame. This motivates the ability to easily "aggregate" multiple json files into each partition using `dask_cudf.read_json`. This PR introduces the `aggregate_files` argument (defaults to `True`) to make it easier to accomplish multi-file DataFrame partitions.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by rjzamora** [comment]
cc @randerzander @ayushdg

File: python/dask_cudf/dask_cudf/io/tests/test_json.py
Comment (rjzamora) [inline python/dask_cudf/dask_cudf/io/tests/test_json.py:111]:
```suggestion
    df2 = dask_cudf.read_json(json_path, aggregate_files=2)
    assert df2.npartitions * 2 == df1.npartitions
    dd.assert_eq(df1, df2, check_index=False)
```

File: python/dask_cudf/dask_cudf/io/json.py
Comment (rjzamora) [inline python/dask_cudf/dask_cudf/io/json.py:149]: It probably makes sense to resolve this before the PR is merged. Shouldn't be terribly difficult to support, but performance will certainly be degraded.

File: python/dask_cudf/dask_cudf/io/tests/test_json.py
Comment (pentschev) [inline python/dask_cudf/dask_cudf/io/tests/test_json.py:110]: Any chance df1 could lead to an odd number of files and thus aggregating two-by-two would lead to a different number of partitions, e.g., `3 * 2 == 5` (number of files is `5`)?

File: python/dask_cudf/dask_cudf/io/json.py
Comment (pentschev) [inline python/dask_cudf/dask_cudf/io/json.py:47]: Not sure how relevant this may be, but what if the data being read contains a `"path"` column? Perhaps it would make sense to make the name of the column configurable or use a very unlikely name, perhaps `"__cudf_path__"`.

**Review by pentschev — COMMENTED** [review]
Left a few comments, overall it looks fine to me, but I'm not very knowledgeable of the dask-cudf codebase to offer the most insightful feedback.

File: python/dask_cudf/dask_cudf/backends.py
Comment (rjzamora) [reply python/dask_cudf/dask_cudf/backends.py:673]: Yes, it is definitely better practice to use a distinct name. I'll make this change - thanks!

File: python/dask_cudf/dask_cudf/io/tests/test_json.py
Comment (rjzamora) [reply python/dask_cudf/dask_cudf/io/tests/test_json.py:110]: You're right - We never know if/when `dask.datasets.timeseries` behavior will change. I can generalize the assertion to:

```python
assert df2.npartitions == math.ceil(df1.npartitions / 2)
```

**Review by pentschev — APPROVED** [review]
LGTM, once again I'm not very knowledgeable of dask-cudf so I'll leave it up to you @rjzamora to merge it or wait for other reviews. 🙂

**Pull request merged**
